Sources/Tracking/CustomerIO.swift

```diff
@@ -35,15 +35,25 @@ public extension CustomerIO {
         configure configureHandler: ((inout SdkConfig) -> Void)?
     ) {
         var sdkConfig = SdkConfig.Factory.create(siteId: siteId, apiKey: apiKey, region: region)
+        let newDiGraph = DIGraph(sdkConfig: sdkConfig)
+        let implementation = DataPipeline.initialize(moduleConfig: DataPipelineConfigOptions.Factory.create(sdkConfig: sdkConfig))
+        let profileStore = newDiGraph.profileStore
+
+        // This code handles the scenario where a user migrates
+        // from the Journeys module to the CDP module while already logged in.
+        // This ensures the CDP module is informed about the
+        // currently logged-in user for seamless processing of events.
+        if DataPipeline.shared.analytics.userId == nil {
```

Comment:
It would be great if we can write a test case for it too

Reply:
For readability and testability, I think it would be good to move this code into it's own encapsulated migration class.